includes/class-wcpdf-admin.php

```diff
@@ -530,77 +530,11 @@ public function save_invoice_number_date($post_id, $post) {
 			if ( ! isset($_POST['action']) || ! in_array($_POST['action'], array( 'editpost', 'wpo_wcpdf_regenerate_document' )) ) {
 				return;
 			}
-
-			// grab json data from the action 'wpo_wcpdf_regenerate_document' and save it to the $_POST variable to be saved
-			if( isset($_POST['json_data']) && ! empty($_POST['json_data']) ) {
-				$form_data = json_decode(stripslashes( $_POST['json_data'] ), true);
-				foreach( $form_data as $key => $value ) {
-					$_POST[$key] = sanitize_text_field($value);
-				}
-			}
 			
 			$order = WCX::get_order( $post_id );
 			if ( $invoice = wcpdf_get_invoice( $order ) ) {
-				if ( !empty( $_POST['wcpdf_invoice_date'] ) ) {
-					$date = $_POST['wcpdf_invoice_date'];
-					$hour = !empty( $_POST['wcpdf_invoice_date_hour'] ) ? $_POST['wcpdf_invoice_date_hour'] : '00';
-					$minute = !empty( $_POST['wcpdf_invoice_date_minute'] ) ? $_POST['wcpdf_invoice_date_minute'] : '00';
-
-					// clean & sanitize input
-					$date = date( 'Y-m-d', strtotime( $date ) );
-					$hour = sprintf('%02d', intval( $hour ));
-					$minute = sprintf('%02d', intval( $minute ) );
-					$invoice_date = "{$date} {$hour}:{$minute}:00";
-
-					// set date
-					$invoice->set_date( $invoice_date );
-				} elseif ( empty( $_POST['wcpdf_invoice_date'] ) && !empty( $_POST['_wcpdf_invoice_number'] ) ) {
-					$invoice->set_date( current_time( 'timestamp', true ) );
-				}
-
-				if ( isset( $_POST['_wcpdf_invoice_number'] ) ) {
-					// sanitize
-					$invoice_number = sanitize_text_field( $_POST['_wcpdf_invoice_number'] );
-					// set number
-					$invoice->set_number( $invoice_number );
-				}
-
-				if ( isset( $_POST['wcpdf_invoice_notes'] ) ) {
-					// allowed HTML
-					$allowed_html = array(
-						'a'		=> array(
-							'href' 	=> array(),
-							'title' => array(),
-							'id' 	=> array(),
-							'class'	=> array(),
-							'style'	=> array(),
-						),
-						'br'	=> array(),
-						'em'	=> array(),
-						'strong'=> array(),
-						'div'	=> array(
-							'id'	=> array(),
-							'class' => array(),
-							'style'	=> array(),
-						),
-						'span'	=> array(
-							'id' 	=> array(),
-							'class'	=> array(),
-							'style'	=> array(),
-						),
-						'p'		=> array(
-							'id' 	=> array(),
-							'class' => array(),
-							'style' => array(),
-						),
-						'b'		=> array(),
-					);
-					// sanitize
-					$invoice_notes = wp_kses( stripslashes($_POST['wcpdf_invoice_notes']), $allowed_html );
-					// set notes
-					$invoice->set_notes( $invoice_notes );
-				}
-
+				$document_data = $this->process_order_document_form_data( $_POST );
```

Comment:
I don't think this will work because `$_POST` will not have `form_data` set (see comment below)